### DISCUSSION TITLE
怎么实现跨视图控制显示与隐藏？

### DISCUSSION THREAD

**Discussion by terrihonkai** [opensumi/core#2668] [Q&A]
![image](https://user-images.githubusercontent.com/132564832/236176635-a41b2ffa-b182-4d2b-a121-dc1f5cccde83.png)

怎么实现topbar去控制两侧的展开与隐藏，还有两侧视图的原有icon隐藏呢？（类似小程序开发者工具的布局）

顺便问一个样式问题，copy了opensumi default theme仓库的light.json样式覆盖，发现editor的样式会有问题，没完全覆盖，会出现这种情况

![image](https://user-images.githubusercontent.com/132564832/236176968-e7094491-e39c-46f3-9404-53cbcd06c843.png)

**Comment by bytemain** [comment]
第一个问题可以看下这个例子：
https://github.com/opensumi/opensumi-module-samples/pull/26

**Reply by terrihonkai** [reply]
感谢，那第二个问题呢，主要是我希望可以单独设置默认样式是明亮的，于是我找了light.json覆盖，但是发现有的地方覆盖不完全

**Reply by terrihonkai** [reply]
试了下确实可以控制视图了，但是像一楼图中那种隐藏掉tab栏是怎么操作的呢？就是不能用tab栏切换，只能用按钮去切换每个tab

**Comment by erha19** [comment]
@terrihonkai 如果想修改主题，需要使用主题包的方式去设置，不能直接进行样式覆盖，如在集成时设置默认主题为:

```ts
const appConfig = {
  ...
  defaultPreferences: {  'general.theme': '亮主题名'},
  ...
}
```

**Reply by terrihonkai** [reply]
那就是我打包的时候要把默认的一些样式插件打进去，然后代码里配好默认的样式主题？那怎么把一些默认的插件打进去呢

**Reply by erha19** [reply]
如果你使用的是 `ide-electron` 项目，你可以自行修改插件下载脚本，追加插件到列表内，这样插件就会下载到 `./extensions` 相对路径，默认会打包进客户端资源，其他项目类似。

**Reply by terrihonkai** [reply]
> 如果你使用的是 `ide-electron` 项目，你可以自行修改插件下载脚本，追加插件到列表内，这样插件就会下载到 `./extensions` 相对路径，默认会打包进客户端资源，其他项目类似。

这个脚本有具体路径吗？确实是ide-electron项目

我只看到build/extensions.json，还不知道怎么追加到列表

**Reply by erha19** [reply]
是这个文件声明的，对应的插件可以到 https://marketplace.opentrs.cn/square 查询并添加你需要的插件。
